core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -42,25 +43,18 @@
  * A component to serve http requests, backed by rest handlers.
  */
 public class HttpServer extends AbstractLifecycleComponent<HttpServer> implements HttpServerAdapter {
-
-    private final Environment environment;
-
     private final HttpServerTransport transport;
 
     private final RestController restController;
 
     private final NodeService nodeService;
 
     @Inject
-    public HttpServer(Settings settings, Environment environment, HttpServerTransport transport,
-                      RestController restController,
-                      NodeService nodeService) {
+    public HttpServer(Settings settings, HttpServerTransport transport, RestController restController, NodeService nodeService) {
         super(settings);
-        this.environment = environment;
         this.transport = transport;
         this.restController = restController;
         this.nodeService = nodeService;
-        nodeService.setHttpServer(this);
```

Comment:
As I created `HttpServerTests` I noticed that `HttpServer` has a mutual dependency on `NodeService` which is a very heavyweight dependency and made it hard to test `HttpServer`. I did not want to get rid of this dependency within the scope of this PR but still reduce the coupling a bit. That's why I removed this line and let Guice inject the HttpServer by adding `setHttpServer` (similarly to the already existing method `setScriptService`). After this PR is done I want to look more closely whether I can get rid of this hard mutual dependency alltogether. And finally: Wiring code like this line belongs to infrastructure code and clutters domain classes.